plugins/inputs/elasticsearch/elasticsearch.go

```diff
@@ -17,71 +16,15 @@ import (
 	"strings"
 )
 
-// mask for masking username/password from error messages
-var mask = regexp.MustCompile(`https?:\/\/\S+:\S+@`)
-
-// Nodestats are always generated, so simply define a constant for these endpoints
+// Define a constant for the endpoints that are always hit. Optional endpoints are
+// defined where they are performed.
 const statsPath = "/_nodes/stats"
 const statsPathLocal = "/_nodes/_local/stats"
 
-type nodeStat struct {
-	Host       string            `json:"host"`
-	Name       string            `json:"name"`
-	Attributes map[string]string `json:"attributes"`
-	Indices    interface{}       `json:"indices"`
-	OS         interface{}       `json:"os"`
-	Process    interface{}       `json:"process"`
-	JVM        interface{}       `json:"jvm"`
-	ThreadPool interface{}       `json:"thread_pool"`
-	FS         interface{}       `json:"fs"`
-	Transport  interface{}       `json:"transport"`
-	HTTP       interface{}       `json:"http"`
-	Breakers   interface{}       `json:"breakers"`
-}
-
-type clusterHealth struct {
-	ClusterName         string                 `json:"cluster_name"`
-	Status              string                 `json:"status"`
-	TimedOut            bool                   `json:"timed_out"`
-	NumberOfNodes       int                    `json:"number_of_nodes"`
-	NumberOfDataNodes   int                    `json:"number_of_data_nodes"`
-	ActivePrimaryShards int                    `json:"active_primary_shards"`
-	ActiveShards        int                    `json:"active_shards"`
-	RelocatingShards    int                    `json:"relocating_shards"`
-	InitializingShards  int                    `json:"initializing_shards"`
-	UnassignedShards    int                    `json:"unassigned_shards"`
-	Indices             map[string]indexHealth `json:"indices"`
-}
-
-type indexHealth struct {
-	Status              string `json:"status"`
-	NumberOfShards      int    `json:"number_of_shards"`
-	NumberOfReplicas    int    `json:"number_of_replicas"`
-	ActivePrimaryShards int    `json:"active_primary_shards"`
-	ActiveShards        int    `json:"active_shards"`
-	RelocatingShards    int    `json:"relocating_shards"`
-	InitializingShards  int    `json:"initializing_shards"`
-	UnassignedShards    int    `json:"unassigned_shards"`
-}
-
-type clusterStats struct {
-	NodeName    string      `json:"node_name"`
-	ClusterName string      `json:"cluster_name"`
-	Status      string      `json:"status"`
-	Indices     interface{} `json:"indices"`
-	Nodes       interface{} `json:"nodes"`
-}
-
-type catMaster struct {
-	NodeID   string `json:"id"`
-	NodeIP   string `json:"ip"`
-	NodeName string `json:"node"`
-}
-
```

Comment:
Lets leave them up top if only to reduce the cognitive load reviewing the diffs.